src/main/java/lotto/domain/LottoStore.java

```diff
@@ -1,59 +1,63 @@
 package lotto.domain;
 
+import lotto.context.Error;
 import lotto.util.LottoNumberUtil;
 
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 public class LottoStore {
-    private static final String TEXT_SPLIT_DIVISION = ",";
     private static final int DEFAULT_RANGE_START = 0;
 
-    private final List<LottoPaper> manualPapers;
-    private final List<LottoPaper> autoPapers;
+    private final LottoPapers manualPapers;
+    private final LottoPapers autoPapers;
 
     public LottoStore(List<String> autoTexts, int paperCount) {
-        manualPapers = makeManualLottoNums(autoTexts);
-        autoPapers = makeAutoLottoNums(paperCount - manualPapers.size());
+        manualPapers = new LottoPapers(makeManualLottoNums(autoTexts));
+        autoPapers = new LottoPapers(makeAutoLottoNums(paperCount - manualPapers.getPapersSize()));
+        validMaxLotto(paperCount);
+    }
+
+    private void validMaxLotto(int paperCount) {
+        if ((getAutoCount() + getManuaCount()) > paperCount) {
+            throw new IllegalArgumentException(Error.ERROR_LOTTO_MAX_OVER.getMsg());
+        }
+    }
+
+    private List<LottoPaper> makeManualLottoNums(List<String> autoTexts) {
+        if (autoTexts == null) {
+            return new ArrayList<>();
+        }
+        List<LottoPaper> lottoPapers = new ArrayList<>();
+        for (String text : autoTexts) {
+            lottoPapers.add(new LottoPaper(text));
+        }
+        return lottoPapers;
     }
 
     private List<LottoPaper> makeAutoLottoNums(Integer paperCount) {
         return IntStream.range(DEFAULT_RANGE_START, paperCount)
-                .mapToObj(value -> new LottoPaper(getAutoLottoNums()))
+                .mapToObj(value -> new LottoPaper(makeAutoLottoNums()))
                 .collect(Collectors.toList());
     }
 
-    private List<LottoNum> getAutoLottoNums() {
+    private List<LottoNum> makeAutoLottoNums() {
         return LottoNumberUtil.getAutoLottoNums();
     }
 
-    private List<LottoPaper> makeManualLottoNums(List<String> autoTexts) {
-        if (autoTexts == null) return new ArrayList<>();
-        List<LottoPaper> lottoPapers = new ArrayList<>();
-        for (String text : autoTexts) {
-            lottoPapers.add(new LottoPaper(Arrays.stream(text.split(TEXT_SPLIT_DIVISION))
-                    .map(String::trim)
-                    .map(s -> LottoNum.of(s))
-                    .collect(Collectors.toList())));
-        }
-        return lottoPapers;
+    public LottoPapers makeLottoPapers() {
+        return LottoPapers.makeAll(new LottoPapers[]{manualPapers, autoPapers});
```

Comment:
두개의 복권을 하나로 합치기 위해 static 메서드와 배열을 사용하게 된 것 같네요!
다음과 같이 객체에게 메시지 보내는식으로 구현하여 수동 복권과 자동 복권을 합치는건 어떨까요 :) ?
```suggestion
        return manualPapers.addAll(autoPapers);
```

Reply:
파라미터로 한번에 받아서 처리하면 깔끔할줄 알았는데 더 복잡해졌네요
그래도 한가지 궁금한게 있습니다
만약 LottoPapers객체가 3가지 이상이 파라미터로 넘어가서 하나로 만들어줘야 하는 경우에는 
`return LottoPapers.makeAll(new LottoPapers[]{papers1, papers2, papers3});`
제가 사용한 방식처럼 이렇게 사용해도 좋을까요?

Reply:
파라미터의 갯수가 불규칙적이라면 말씀해주신대로 배열을 받도록 하고, 수행하는 메서드를 객체의 오퍼레이션으로 취급하는 것 보다 유틸성으로 관리하는게 더 효율적일 것 같아요 :)